File: src/main/java/one/devos/nautical/SofterPastels/SofterPastels.java
package one.devos.nautical.SofterPastels;

import gay.asoji.fmw.FMW;
import net.devtech.arrp.api.RRPCallback;
import net.devtech.arrp.api.RuntimeResourcePack;
import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.itemgroup.v1.FabricItemGroup;
import net.fabricmc.loader.api.FabricLoader;
import net.minecraft.core.Registry;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.ItemStack;
import one.devos.nautical.SofterPastels.common.SofterPastelsBlocks;
import one.devos.nautical.SofterPastels.common.SofterPastelsItems;
import one.devos.nautical.SofterPastels.common.SofterPastelsTab;
import one.devos.nautical.SofterPastels.common.datagen.Blockstate;
import one.devos.nautical.SofterPastels.common.datagen.LootTables;
import one.devos.nautical.SofterPastels.common.datagen.Recipes;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class SofterPastels implements ModInitializer {
    public static final String MOD_ID = "softerpastels";
    public static final String VERSION = FMW.getVersionString(MOD_ID);
    public static final String MOD_NAME = FMW.getName(MOD_ID);
    public static final RuntimeResourcePack RUNTIME_RESOURCE_PACK = RuntimeResourcePack.create(new ResourceLocation(MOD_ID, "arrp"));

    public static final Logger LOGGER = LogManager.getLogger(MOD_NAME);

    public static ResourceLocation locate(String location) {
        if (location.contains(":")) {
            return new ResourceLocation(location); //probably useful if its from another mod
        } else {
            return new ResourceLocation(MOD_ID, location); //so we can be lazy, or if its not from another mod
        }
    }

    public static final CreativeModeTab SP_ITEM_GROUP = FabricItemGroup.builder()
            .icon(() -> new ItemStack(SofterPastelsBlocks.RED_PASTEL_BLOCK))
            .title(Component.translatable("itemGroup.softerpastels"))
            .displayItems(SofterPastelsTab::build)
            .build();

    @Override
    public void onInitialize() {
        SofterPastelsBlocks.init();
        SofterPastelsItems.init();

        Registry.register(BuiltInRegistries.CREATIVE_MODE_TAB, locate("main"), SP_ITEM_GROUP);

        Blockstate.init();
        LootTables.init();
        Recipes.init();

        RRPCallback.EVENT.register(a -> a.add(SofterPastels.RUNTIME_RESOURCE_PACK));

        LOGGER.info("[" + MOD_NAME + "] Version " + VERSION + " loaded.");
        LOGGER.info("[" + MOD_NAME + "] " + "Getting ready to load some soft sweet eye candy.");
    }
}

File: src/main/kotlin/one/devos/nautical/softerpastels/SofterPastels.kt
package one.devos.nautical.softerpastels

import gay.asoji.fmw.FMW
import gay.asoji.innerpastels.InnerPastels
import net.fabricmc.api.ModInitializer
import net.fabricmc.fabric.api.itemgroup.v1.FabricItemGroup
import net.fabricmc.loader.api.FabricLoader
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.network.chat.Component
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.item.CreativeModeTab
import net.minecraft.world.item.ItemStack
import one.devos.nautical.softerpastels.common.SofterPastelsBlocks
import one.devos.nautical.softerpastels.common.SofterPastelsItems
import one.devos.nautical.softerpastels.common.SofterPastelsTab
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

object SofterPastels : ModInitializer {
    const val MOD_ID: String = "softerpastels"
    val VERSION: String = FMW.getVersionString(MOD_ID)
    val MOD_NAME: String = FMW.getName(MOD_ID)

    val LOGGER: Logger = LogManager.getLogger(MOD_NAME)

    // could prob be ridded of tbh, the only place this has ever been used is line 44... i dont even know why this exists, legacy codebase?
    fun locate(location: String): ResourceLocation {
        return if (location.contains(":")) {
            ResourceLocation(location) //probably useful if its from another mod
        } else {
            ResourceLocation(MOD_ID, location) //so we can be lazy, or if its not from another mod
        }
    }

    val SP_ITEM_GROUP: CreativeModeTab = FabricItemGroup.builder()
        .icon { ItemStack(SofterPastelsBlocks.RED_PASTEL_BLOCK) }
        .title(Component.translatable("itemGroup.softerpastels"))
        .displayItems(SofterPastelsTab::build)
        .build()

    override fun onInitialize() {
        SofterPastelsBlocks.init()
        SofterPastelsItems.init()

        Registry.register(BuiltInRegistries.CREATIVE_MODE_TAB, locate("main"), SP_ITEM_GROUP)

        InnerPastels.registerMods(MOD_ID)

        if (!FabricLoader.getInstance().isModLoaded("desolatedpastels")) { // Reduce a little bit of log spam if Desolated is loaded, otherwise, run.
            LOGGER.info("[${MOD_NAME}] Version ${VERSION} loaded.")
            LOGGER.info("[${MOD_NAME}] Internal Library Version ${FMW.getVersionString("innerpastels")} loaded. Please include this when reporting bugs!!!")
            LOGGER.info("[${MOD_NAME}] Getting ready to load some soft sweet eye candy.")
        }

    }
}